app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -477,6 +484,35 @@ def parse_stack_resource(resource, group)
         return uid, new_result
       end
 
+      def parse_managed_image(image)
+        uid = image.id.downcase
+
+        resource_group = ResourceGroup.where(:name => image.resource_group.downcase).find do |rg|
+          rg.ems_ref.include?(@ems.subscription)
+        end
+
+        new_result = {
+          :type               => ManageIQ::Providers::Azure::CloudManager::Template.name,
+          :uid_ems            => uid,
+          :ems_ref            => uid,
+          :name               => image.name,
+          :description        => "#{image.resource_group}\\#{image.name}",
+          :location           => @ems.provider_region,
+          :vendor             => 'azure',
+          :raw_power_state    => 'never',
+          :template           => true,
+          :publicly_available => false,
+          :resource_group_id  => resource_group.try(:id),
+          :operating_system   => process_os(image),
+          :hardware           => {
+            :bitness  => 64,
+            :guest_os => image.properties.storage_profile.os_disk.os_type
```

Comment:
I wonder if we could use few try() here, can it be the specific properties are not always there?